src/Core/Domain/ImageSettings/CommandHandler/RegenerateThumbnailsHandler.php

```diff
@@ -54,59 +57,89 @@ public function __construct(
      * {@inheritdoc}
      *
      * @throws ImageTypeException
+     * @throws RegenerateThumbnailsException
      */
     public function handle(RegenerateThumbnailsCommand $command): void
     {
         // Get all languages
         $languages = $this->langRepository->findAll();
 
-        // Set images to regenerate with all theirs specific directories
-        $process = [
-            ['type' => 'categories', 'dir' => _PS_CAT_IMG_DIR_],
-            ['type' => 'manufacturers', 'dir' => _PS_MANU_IMG_DIR_],
-            ['type' => 'suppliers', 'dir' => _PS_SUPP_IMG_DIR_],
-            ['type' => 'products', 'dir' => _PS_PRODUCT_IMG_DIR_],
-            ['type' => 'stores', 'dir' => _PS_STORE_IMG_DIR_],
-        ];
+        $image = ImageDomain::from($command->getImage());
+        $imageTypeId = null;
+        if ($command->getImageTypeId() > 0) {
+            $imageTypeId = new ImageTypeId($command->getImageTypeId());
+        }
 
-        // Launching generation process
-        foreach ($process as $proc) {
+        // Iterate all images categories with thumbnails
+        foreach (ImageDomain::getDomainsWithThumbnails() as $imageCategory) {
```

Comment:
```suggestion
        foreach (ImageDomain::getDomainsWithThumbnails() as $imageDomain) {
```
Minor suggestion but I believe in your previous implementation `ImageDomain` class was named `ImageCategory` right? So the variable name probably comes from this previous implementation, it would be easier to read if the variable reflects the class IMHO